Eastern-Research-Group/mywaterway
# Feature/hmw 15 explore clipping waterbodies

## Related Issues:
* https://jira.epa.gov/browse/HMW-15

## Main Changes:
* Added logic to clip waterbodies to within the selected huc.
* Added logic to highlight the entire waterbody, including parts that extend outside the selected huc, when the user mouses over or selects a waterbody.

## Steps To Test:
1. Navigate dc
2. Verify the waterbodies don't extend outside of the huc.
3. Verify there is no green displaying for this particular huc.
4. Mouse over some waterbodies that should extend outside of the huc and verify the entire waterbody is highlighted
5. Click a view on map button for a waterbody that extends outside of the huc and verify the entire waterbody is highlighted

## TODO:
- [x] Get client feedback and implement any changes
- [x] Merge #506 prior to merging this PR, as this one is based on #506
